doc/development.md

```diff
@@ -120,3 +120,180 @@
 
 Before running `make` you could use `./configure` to set some (rarely used) options.
 To see the list of options: `./configure -h`.
+
+
+## Coding guidelines
+
+The coding conventions in Vdsm are generally aligned with the
+[PEP 8](https://peps.python.org/pep-0008/) style recommendations.
+Code is linted with `flake8` and `pylint` based in the `/.pylintrc` file.
+You can trigger the linters check by pushing changes into your fork, which invokes:
+
+    make lint
+
+However, some code conventions are not explicitly enforced by the linters due
+to their limitations. Nonetheless, new code shall adhere to those
+rules, and old code should be updated as possible when patched.
+
+In general, is very important for new code to be consistent with the existing
+code in the same module, or in the same package for new modules. In case of
+doubt, check the already existing code before!
+
+In this section, we will cover some of the most important conventions and
+preferences for contributing to Vdsm, as extension to PEP 8 general rules.
+
+### Imports
+
+First import standard library modules, then 3rd party modules, and finally
+project modules.
+Separate standard library modules and 3rd party modules with a blank line.
+Keep alphabetical order within each block.
+
+Less specific imports:
+
+    from vdsm.foo import baz
+
+Should come before more specific imports like:
+
+    from vdsm.foo.bar import baz
+
+Relative name imports are accepted, but better to avoid.
+
+### Indentation
+
+Use 4 spaces per indentation level.
+
+Continuation lines should align wrapped elements either vertically using
+Python's implicit line joining inside parentheses, brackets and braces, or
+using a hanging indent. When using a hanging indent, the following should
+be considered:
+- There should be no arguments on the first line.
+- Further indentation should be used to clearly distinguish itself as a continuation line.
+- Hanging indentation should stick to the 4-space rule.
+```python
+# Add 4 spaces (an extra level of indentation) to distinguish arguments from the rest.
+# Consider that having too many arguments in a function is
+foo = long_function_name(
+    var_one, var_two, var_three, var_four)
+
+# Recommended extra indentation with multiline if-statements.
+if (condition_1 and condition_2 [...]
+        and condition_n):
+    do_something()
+
+# But this is also possible.
+def long_function_name(
+        var_one, var_two, var_three,
+        var_four):
+    print(var_one)
+```
+
+To break comprehension Lists, Dicts, Sets as well as generator expressions,
+keep expressions that would be written in a single line together.
+Continuation lines stick to the same rules as above. Closing
+brace/bracket/parenthesis shall line up under the first character of the
+line start.
```

Comment:
But in the following example the closing bracket is at the end of line.  Which is all right, just confusing with respect to this sentence.  It might be clarified by saying e.g. "Closing brace/bracket/parenthesis, if placed on a separate line, shall line up ...".